src/blockrelay/blockrelay_common.cpp

```diff
@@ -216,70 +224,75 @@ bool ThinTypeRelay::AddBlockInFlight(CNode *pfrom, const uint256 &hash, const st
     if (AreTooManyBlocksInFlight())
         return false;
 
-    // Verify that we haven't already added this entry before since mapThinTypeBlocksInFlight
-    // is a multi-map and we could end up adding two identical entries.
-    if (!IsBlockInFlight(pfrom, thinType, hash))
+    // this insert returns a pair <iterator,bool> where the bool denotes whether the insertion took place
+    auto key = mapThinTypeBlocksInFlight.find(pfrom->GetId());
+    if (key != mapThinTypeBlocksInFlight.end())
     {
-        mapThinTypeBlocksInFlight.insert(
-            std::pair<const NodeId, CThinTypeBlockInFlight>(pfrom->GetId(), {hash, GetTime(), false, thinType}));
-        return true;
+        return key->second.emplace(CThinTypeBlockInFlight{hash, GetTime(), false, thinType}).second;
```

Comment:
This section of code is hard to read...why not mostly leave it the way it was.... if the IsBlockInFlight() is not true then add it, seems much simpler and easier to read.

Reply:
It is doing that. But we cant just call IsBlockInFlight because we dont have enough information. 

IsBlockInFlight will only tell us if there is a block in flight for this peer. but we also need to know there is an entry in the map for this peer since that changes the way we add the data. the fitst if statement does the check to see if there is an entry for this peer in the map, if there is; add the block. if there isnt; then add an entry for this peer then add the block.

Reply:
ok good